realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -346,14 +313,56 @@ protected void onError(LogoutResponse response) {
     }
 
     /**
+     * Change this users password. This is done synchronously and involves the network, so calling this method on the
+     * Android UI thread will always crash.
+     *
+     * @param newPassword the users new password.
+     * @throws ObjectServerError if the password could not be changed for some reason.
+     */
+    public void changePassword(String newPassword) throws ObjectServerError {
+        if (newPassword == null) {
```

Comment:
can you use an empty password? what's the password requirement from the server POV? (number of char etc.)
or do you rely on the server returning the error about the size?